plugins/modules/system/sap_snote.py

```diff
@@ -0,0 +1,253 @@
+    type: list
+    elements: dict
+    returned: always
+    sample: '{
+        "RETURN": [{"ES_MSG": { "MSGNO": "000", "MSGTY": "", "MSGTXT": "", "MSGV1": "" },
+                "ET_MSG": [],
+                "EV_RC": 0,
+                "ET_MISSING_NOTES": [],
+                "IT_FILENAME": [{"FILENAME": "/usr/sap/trans/tmp/0002980265.txt"}],
+                "IT_NOTES": [{"NUMM": "0002980265", "VERSNO": "0000"}]
+                }]}'
+'''
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+import traceback
+try:
+    from pyrfc import Connection
+except ImportError:
+    HAS_PYRFC_LIBRARY = False
+    ANOTHER_LIBRARY_IMPORT_ERROR = traceback.format_exc()
+else:
+    HAS_PYRFC_LIBRARY = True
+
+
+def call_rfc_method(connection, method_name, kwargs):
+    # PyRFC call function
+    return connection.call(method_name, **kwargs)
+
+
+def check_implementation(conn, snote):
+    check_implemented = call_rfc_method(conn, 'SCWB_API_GET_NOTES_IMPLEMENTED', {})
+    for snote_list in check_implemented['ET_NOTES_IMPL']:
+        if snote in snote_list['NUMM']:
+            return True
+    return False
+
+
+def run_module():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['absent', 'present']),
+            conn_username=dict(type='str', required=True),
+            conn_password=dict(type='str', required=True, no_log=True),
+            host=dict(type='str', required=True),
+            sysnr=dict(type='str', default="01"),
+            client=dict(type='str', default="000"),
+            snote_path=dict(type='str', required=False),
+            snote=dict(type='str', required=False),
+        ),
+        required_one_of=[('snote_path', 'snote')],
+        supports_check_mode=False,
+    )
+    result = dict(changed=False, msg='', out={}, error='')
+    raw = ""
+    post_check = False
+
+    params = module.params
+
+    state = params['state']
+    conn_username = (params['conn_username']).upper()
+    conn_password = params['conn_password']
+    host = params['host']
+    sysnr = params['sysnr']
+    client = params['client']
+
+    path = params['snote_path']
+    snote = params['snote']
+
+    if not HAS_PYRFC_LIBRARY:
+        module.fail_json(
+            msg=missing_required_lib('pyrfc'),
+            exception=ANOTHER_LIBRARY_IMPORT_ERROR)
+
+    # basic RFC connection with pyrfc
+    try:
+        conn = Connection(user=conn_username, passwd=conn_password, ashost=host, sysnr=sysnr, client=client)
+    except Exception as err:
+        result['error'] = str(err)
+        result['msg'] = 'Something went wrong connecting to the SAP system.'
+        module.fail_json(**result)
+
+    # pre evaluation of parameters
+    if path is not None:
+        if path.endswith('.txt'):
+            # splits snote number from path and txt extension
+            snote = path.split('/')[-1].split('.')[0]
```

Comment:
I guess we should take a look into the os.path methods to parse paths

Reply:
Would be a bit more sophisticated